modules/dnn/src/model.cpp

```diff
@@ -0,0 +1,256 @@
+        // [batchId, classId, confidence, left, top, right, bottom]
+        for (int i = 0; i < detections.size(); ++i)
+        {
+            float* data = (float*)detections[i].data;
+            for (int j = 0; j < detections[i].total(); j += 7)
+            {
+                float conf = data[j + 2];
+                if (conf < confThreshold)
+                    continue;
+
+                int left   = data[j + 3];
+                int top    = data[j + 4];
+                int right  = data[j + 5];
+                int bottom = data[j + 6];
+                int width  = right  - left + 1;
+                int height = bottom - top + 1;
+
+                if (width * height <= 1)
+                {
+                    left   = data[j + 3] * frameWidth;
+                    top    = data[j + 4] * frameHeight;
+                    right  = data[j + 5] * frameWidth;
+                    bottom = data[j + 6] * frameHeight;
+                    width  = right  - left + 1;
+                    height = bottom - top + 1;
+                }
+
+                left   = std::max(0, std::min(left, frameWidth - 1));
+                top    = std::max(0, std::min(top, frameHeight - 1));
+                width  = std::max(0, std::min(width, frameWidth - 1 - left));
+                height = std::max(0, std::min(height, frameHeight - 1 - top));
+                predBoxes.emplace_back(left, top, width, height);
+
+                predClassIds.push_back(static_cast<int>(data[j + 1]));
+                predConf.push_back(conf);
+            }
+        }
+    }
+    else if (lastLayer->type == "Region")
+    {
+        for (int i = 0; i < detections.size(); ++i)
+        {
+            // Network produces output blob with a shape NxC where N is a number of
+            // detected objects and C is a number of classes + 4 where the first 4
+            // numbers are [center_x, center_y, width, height]
+            float* data = (float*)detections[i].data;
+            for (int j = 0; j < detections[i].rows; ++j, data += detections[i].cols)
+            {
+
+                Mat scores = detections[i].row(j).colRange(5, detections[i].cols);
+                Point classIdPoint;
+                double conf;
+                minMaxLoc(scores, nullptr, &conf, nullptr, &classIdPoint);
+
+                if (conf < confThreshold)
+                    continue;
+
+                int centerX = std::max(0, std::min(static_cast<int>(data[0] * frameWidth), frameWidth - 1));
+                int centerY = std::max(0, std::min(static_cast<int>(data[1] * frameHeight), frameWidth - 1));
+                int width   = data[2] * frameWidth;
+                int height  = data[3] * frameHeight;
+
+                int left = std::max(0, std::min(centerX - width / 2, frameWidth - 1));
+                int top  = std::max(0, std::min(centerY - height / 2, frameHeight - 1));
+                width    = std::max(0, std::min(width, frameWidth - 1 - left));
```

Comment:
If `left == 0` and `width ==1` and `frameWidth == 1` we should have valid `left == 0` and `width == 1`. For now it assigns `width = min(1, 1 - 1 - 0) = 0`.